lib/stripe/resources/account.rb

```diff
@@ -45,12 +45,8 @@ def resource_url
     end
 
     # @override To make id optional
-    def self.retrieve(id = ARGUMENT_NOT_PROVIDED, opts = {})
```

Comment:
`ARGUMENT_NOT_PROVIDED` doesn't play very well with how we merge options. The `args` object in `method_missing` is an array and merging options when an ID is omitted results in `[nil, {}]` being sent to the resource, which isn't desirable.

However, I wasn't able to re-create any of the scenarios that warranted the addition of it in https://github.com/stripe/stripe-ruby/commit/c269e9bd9baeee4971c0b760b44cf1212ab26fd6 and believe it's safe for removal.